### PR TITLE
Add modules for core, pipelines, and policy

### PR DESCRIPTION
**Pull request by roanutil**
- Add core module
- Add pipelines module
- Add policy module
- Add workflow config
- Add README
- Add license

### CODE REVIEW

File: Sources/AZPolicyFixtures/PolicyType.swift [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+import AZCore
+import AZCoreFixtures
+import AZPolicy
+import Foundation
+
+// swiftlint:disable line_length
+extension AZFixtures.PolicyType {
+    public static let secretsScanningRestriction = AZPolicyType(
+        id: "ec003f37-8db0-4e10-992a-a2895045752c",
```
Comment (r-jarvis) [inline Sources/AZPolicyFixtures/PolicyType.swift:17]: I'm really unfamiliar with the Azure DevOps, but I noticed quite a few floating IDs here. Are these safe to expose in an OSS project?

Comment (roanutil) [reply Sources/AZPolicyFixtures/PolicyType.swift:17]: Those individual resource IDs should not be sensitive. I did make sure to scrub all references to any real organization or project to be extra safe.

File: Sources/AZCore/AZCreateRequest.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        try? encoder.encode(body)
+    }
+
+    public init(apiVersion: AZApiVersion, body: T) {
+        self.apiVersion = apiVersion
+        self.body = body
+    }
+
+    public static func sevenDotZero(body: T) -> Self {
+        AZCreateRequest(apiVersion: "7.0", body: body)
```
Comment (raj3m8) [inline Sources/AZCore/AZCreateRequest.swift:31]: Should we be passing raw String here or use the AZApiVersion?

File: Sources/AZPolicy/AZPolicyRoute+ConfigurationsRoute.swift [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        }
+    }
+}
+
+extension AZPolicyRoute.ConfigurationsRoute {
+    public typealias DeleteRequest = GetRequest
+
+    public struct GetRequest: Hashable, Sendable, AZVersionedRequest_7_0, AZGetRequest {
+        public let configurationId: AZPolicyConfiguration.ID
+        public let apiVersion: String = "7.0"
```
Comment (r-jarvis) [inline Sources/AZPolicy/AZPolicyRoute+ConfigurationsRoute.swift]: Should we be passing raw String here or use the AZApiVersion?

File: Sources/AZPipelines/AZNewPipeline.swift [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    public let configuration: AZPipeline.Configuration
+
+    public init(name: AZPipeline.Name, folder: String, configuration: AZPipeline.Configuration) {
+        self.name = name
+        self.folder = folder
+        self.configuration = configuration
+    }
+}
+
+extension AZNewPipeline: Identifiable {
```
Comment (r-jarvis) [inline Sources/AZPipelines/AZNewPipeline.swift:24]: Is this safe to use name as an identifier? Could be duplicates

File: Sources/AZPipelines/AZPipeline.swift [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        public let type: RepositoryType
+
+        public init(id: AZRepositoryId, type: RepositoryType) {
+            self.id = id
+            self.type = type
+        }
+    }
+
+    public enum RepositoryType: String, Hashable, Sendable, Codable {
+        case azureReposGit
```
Comment (r-jarvis) [inline Sources/AZPipelines/AZPipeline.swift:88]: Are there potentially more types in the future?

File: Sources/AZCore/AZLink.swift [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+// azure-devops-swift
+//
+// Copyright © 2023 MFB Technologies, Inc. All rights reserved.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+import Foundation
+
+public struct AZReferenceLink: Hashable, Sendable, Codable {
```
Comment (r-jarvis) [inline Sources/AZCore/AZLink.swift]: Was there a specific reasoning for using this struct over using URL() directly in the code?

**Review by r-jarvis — COMMENTED** [review]
Just a few questions, but I think it looks good for the init commit.

File: Sources/AZPipelines/AZPipeline.swift (hunk 5 above)
Comment (roanutil) [reply Sources/AZPipelines/AZPipeline.swift:88]: Possibly. I think it's safer to assume this type can grow with future API versions.

File: Sources/AZCore/AZLink.swift (hunk 6 above)
Comment (roanutil) [reply Sources/AZCore/AZLink.swift]: In the JSON schema it actually has the "href" key for the URL value. For encode/decode, we need our type to match that schema.

**Review by r-jarvis — APPROVED** [review]

**Pull request merged**